### ISSUE TITLE
Post [...] menu options go off the edge of the screen in search results

### ISSUE THREAD

**Issue by brookmiles** [mastodon/mastodon#31737]
### Steps to reproduce the problem

1. Search for a (moderately long) post using the search box
2. Adjust zoom level so the bottom of the post reaches near the bottom of the browser window
3. Click the [...] menu on the post

### Expected behaviour

menu open upwards allowing access to all menu options

### Actual behaviour

menu opens downwards, some menu options can't be reached as they are off the screen

### Detailed description

There does not appear to be any way to scroll either the page or menu to access the options off the edge of the browser.

![20240904-073332-firefox](https://github.com/user-attachments/assets/570f422b-783a-4d8b-951d-583db8a44232)

This only appears to happen while viewing a single post on the search results screen. When viewing the post page directly, the menu opens up or down as appropriate to fit on the screen.

### Mastodon instance

mastodon.social

### Mastodon version

v4.3.0-nightly.2024-09-03

### Browser name and version

Firefox 129.0.2

### Operating system

Microsoft Windows [Version 10.0.19045.4780]

**Comment by brookmiles** [comment]
> This is curious and I'm unable to reproduce it. My hunch is that maybe the menu is too large to fit, even if it were displayed above the “…”?

You're right, that seems to be the case. Zooming in the page far enough that the menu doesn't fit will cause it to open "down", potentially becoming entirely hidden, whereas opening "up" would at least show most of it...

Bit of an edge case I admit... still might be nice to have it open "up" in that situation?

**Comment by ClearlyClaire** [comment]
We're using Popper with the [`flip`](https://popper.js.org/docs/v2/modifiers/flip/) modifier; it doesn't seem to have an option to chose “the least bad placement”.

We could have a look how `preventOverflow` works…
